crates/rpc-types-engine/src/lib.rs

```diff
@@ -18,5 +18,154 @@
 
 extern crate alloc;
 
-mod payload;
-pub use payload::*;
+pub use alloy_rpc_types_engine::ExecutionPayloadV1;
+pub use op_alloy_genesis::{RollupConfig, SystemConfig};
+pub use op_alloy_protocol::L2BlockInfo;
+
+use alloy_eips::eip2718::Decodable2718;
+use alloy_primitives::B256;
+use op_alloy_consensus::{OpTxEnvelope, OpTxType};
+use op_alloy_protocol::{BlockInfo, L1BlockInfoBedrock, L1BlockInfoEcotone, L1BlockInfoTx};
+
+mod attributes;
+pub use attributes::{OptimismAttributesWithParent, OptimismPayloadAttributes};
+
+mod payload_v3;
+pub use payload_v3::OptimismExecutionPayloadEnvelopeV3;
+
+mod payload_v4;
+pub use payload_v4::OptimismExecutionPayloadEnvelopeV4;
+
+mod errors;
+pub use errors::{ToL2BlockRefError, ToSystemConfigError};
+
+/// Converts a payload into its most inner [ExecutionPayloadV1].
+pub trait IntoInnerPayload {
+    /// Converts the payload into its most inner [ExecutionPayloadV1].
+    fn inner_payload(&self) -> &crate::ExecutionPayloadV1;
```

Comment:
can we keep versioning in function name if this returns V1

we can also return `Cow` here so that it's also possible to create this type if this can't be returned as `&`